### PR TITLE
Update injectRecaptchaFields to inject recaptcha enterprise fields into phone API requests

### PR DESCRIPTION
**Pull request by NhienLam**
### Discussion
This is the pre-req for reCAPTCHA Enterprise and phone auth integration support.

- Add Phone provider and action names
- Add reCAPTCHA Enterprise fields to phone API requests
- Update injectRecaptchaFields to inject recaptcha enterprise fields into phone API requests

### Testing
Tested the following flows with the demo app:
- Phone Auth
- SMS MFA
- Email auth with reCAPTCHA Enterprise

### CODE REVIEW

File: packages/auth/src/platform_browser/recaptcha/recaptcha_enterprise_verifier.ts [hunk 1]
```diff
@@ -155,16 +157,61 @@ export async function injectRecaptchaFields<T>(
   auth: AuthInternal,
   request: T,
   action: RecaptchaActionName,
-  captchaResp = false
+  captchaResp = false,
+  fakeToken = false
```
Comment (renkelvin) [inline packages/auth/src/platform_browser/recaptcha/recaptcha_enterprise_verifier.ts:161]: Change them to `isCaptchaResp` and `isFakeToken` which seem better names for boolean values?

File: packages/auth/src/api/account_management/mfa.ts [hunk 2]
```diff
@@ -55,7 +57,10 @@ export interface StartPhoneMfaEnrollmentRequest {
   idToken: string;
   phoneEnrollmentInfo: {
     phoneNumber: string;
-    recaptchaToken: string;
+    recaptchaToken?: string;
+    captchaResponse?: string;
```
Comment (renkelvin) [inline packages/auth/src/api/account_management/mfa.ts:61]: I suggest add comments after each field like "reCAPTCHA v2", "reCAPTCHA Enterprise" for future reference.

File: packages/auth/src/platform_browser/recaptcha/recaptcha.test.ts [hunk 3]
```diff
@@ -46,7 +46,14 @@ describe('platform_browser/recaptcha/recaptcha', () => {
...
     recaptchaKey: 'projects/testproj/keys/' + TEST_SITE_KEY,
     recaptchaEnforcementState: [
-      { provider: 'EMAIL_PASSWORD_PROVIDER', enforcementState: 'ENFORCE' }
+      {
+        provider: RecaptchaProvider.EMAIL_PASSWORD_PROVIDER,
+        enforcementState: EnforcementState.ENFORCE
+      },
+      {
+        provider: RecaptchaProvider.PHONE_PROVIDER,
+        enforcementState: EnforcementState.AUDIT
```
Comment (renkelvin) [inline packages/auth/src/platform_browser/recaptcha/recaptcha.test.ts:55]: Add a combination of `ENFORCE` and `OFF`?

File: packages/auth/src/api/account_management/mfa.ts (hunk 2 above)
Comment (NhienLam) [reply packages/auth/src/api/account_management/mfa.ts]: Done.

File: packages/auth/src/platform_browser/recaptcha/recaptcha.test.ts (hunk 3 above)
Comment (NhienLam) [reply packages/auth/src/platform_browser/recaptcha/recaptcha.test.ts:54]: Done.

File: packages/auth/src/platform_browser/recaptcha/recaptcha_enterprise_verifier.ts (hunk 1 above)
Comment (NhienLam) [reply packages/auth/src/platform_browser/recaptcha/recaptcha_enterprise_verifier.ts]: Done. Thanks!

**Review by renkelvin — APPROVED** [review]

**Pull request merged**